### PR TITLE
Fixes for supporting custom .c test and specific riscv_ebreak_test_0.S

### PR DESCRIPTION
**Pull request by GTumbush**
Fixes for supporting custom .c test
Addition of riscv_ebreak_test_0.S and changes to .mk to support this.
Fixed RISCV_EXE_PREFIX ?= issue in Common.mk

### CODE REVIEW

File: cv32/sim/Common.mk [hunk 1]
```diff
@@ -203,7 +203,7 @@ sanity: hello-world
 # We link with our custom crt0.s and syscalls.c against newlib so that we can
 # use the c standard library
 $(CUSTOM_DIR)/$(CUSTOM_PROG).elf: $(CUSTOM_DIR)/$(CUSTOM_PROG).c
-	$(RISCV_EXE_PREFIX)gcc -march=rv32imc -o $@ -w -Os -g -nostdlib \
+	$(RISCV_EXE_PREFIX)gcc -mabi=ilp32 -march=rv32imc -o $@ -w -Os -g -nostdlib \
```
Comment (MikeOpenHWGroup) [inline cv32/sim/Common.mk:206]: What does `-mabi=ilp32` do and why do we need it?   Does it affect other tests?

File: cv32/sim/Common.mk [hunk 2]
```diff
@@ -262,6 +262,18 @@ $(CUSTOM)/dhrystone.elf: $(CUSTOM)/dhrystone.c
 		-I $(RISCV)/riscv32-unknown-elf/include \
 		-L $(RISCV)/riscv32-unknown-elf/lib \
 		-lc -lm -lgcc
+
```
Comment (MikeOpenHWGroup) [inline cv32/sim/Common.mk:265]: Do we need this?   I would have expected the existing `make custom CUSTOM_PROG=<my_fancy_new_test_proj>` target to handle this.

File: cv32/sim/uvmt_cv32/xrun.mk [hunk 3]
```diff
@@ -113,6 +113,13 @@ dhrystone: comp $(CUSTOM)/dhrystone.hex
 		+UVM_TESTNAME=uvmt_cv32_firmware_test_c \
 		+firmware=$(CUSTOM)/dhrystone.hex
 
+riscv_ebreak_test_0: comp $(CUSTOM)/riscv_ebreak_test_0.hex
```
Comment (MikeOpenHWGroup) [inline cv32/sim/uvmt_cv32/xrun.mk:116]: Same as previous comment.

Comment (GTumbush) [reply cv32/sim/uvmt_cv32/xrun.mk:116]: New code is the ebreak . S test.

When I try to make the riscv_ebreak_test_0 using make custom this is what I get:
make custom CUSTOM_PROG=riscv_ebreak_test_0 SIMULATOR=xrun XRUN_USE_ISS=YES CV32E40P_PKG=/wrk/gtumbush/mcu/OpenHW/cv32e40p/master

make: *** No rule to make target `/wrk/gtumbush/mcu/OpenHW/regression_test/cv32/sim/uvmt_cv32/../../../cv32/tests/core/custom/riscv_ebreak_test_0.c', needed by `/wrk/gtumbush/mcu/OpenHW/regression_test/cv32/sim/uvmt_cv32/../../../cv32/tests/core/custom/riscv_ebreak_test_0.elf'.  Stop.

.S tests don’t seem to be supported as a custom test.

File: cv32/sim/Common.mk (hunk 2 above)
Comment (GTumbush) [reply cv32/sim/Common.mk:265]: This is existing code, only the line feed is new.  The test specific code in this Makefile for hello_world, misalign, illegal, fibonacci, and dhrystone could be removed but then all users need to be aware of the new calling method (make custom CUSTOM_PROG=<my_fancy_new_test_proj>) and the ci script needs updating.

File: cv32/sim/Common.mk (hunk 1 above)
Comment (GTumbush) [reply cv32/sim/Common.mk:206]: mabi specifies integer and floating-point calling convention.

Without this option gcc will error with the following:
c1: error: requested ABI requires -march to subsume the 'D' extension

File: cv32/sim/uvmt_cv32/xrun.mk (hunk 3 above)
Comment (MikeOpenHWGroup) [reply cv32/sim/uvmt_cv32/xrun.mk:116]: Ah!   Good catch.  I'll take an action to figure out how to support the assembler (*.S) custom test-programs.   We'll want that for riscv-dv support for sure.   In the meantime, let's bring in your `riscv_ebreak_test_0` target as a temporary fix.

**Comment by MikeOpenHWGroup** [comment]
Thanks for the quick replies Greg.  I'll merge this in.

**Pull request merged**